theano/tensor/blas.py

```diff
@@ -312,8 +320,31 @@ def default_blas_ldflags():
     # Even if we could not detect what was used for numpy, or if these
     # libraries are not found, most Linux systems have a libblas.so
     # readily available. We try to see if that's the case, rather
-    # than disable blas.
-    if GCC_compiler.try_flags(["-lblas"]):
+    # than disable blas. To test it correctly, we must load a program.
+    # Otherwise, there could be problem in the LD_LIBRARY_PATH.
+
+    test_code = textwrap.dedent("""\
+        extern "C" float sdot_(int*, float*, int*, float*, int*);
+        int main(int argc, char** argv)
+        {
+            int Nx = 5;
+            int Sx = 1;
+            float x[5] = {0, 1, 2, 3, 4};
+            float r = sdot_(&Nx, x, &Sx, x, &Sx);
+
+            if ((r - 30.f) > 1e-6 || (r - 30.f) < -1e-6)
+            {
+                return -1;
+            }
+            return 0;
+        }
+        """)
+    flags = ['-lblas']
+    flags.extend('-L'.join(theano.gof.cmodule.std_lib_dirs()))
```

Comment:
Wait, this line is completely wrong. Just by coincidence it will work if `std_lib_dirs()` is empty. If there are one or more entries, then a) they are concatenated to a single large string with some `-L` in between, and b) they are appended to `flags` character by character. I'll send a PR.